src/csharp/Model.cs

```diff
@@ -42,8 +42,11 @@ protected virtual void Dispose(bool disposing)
             {
                 return;
             }
-            NativeMethods.OgaDestroyModel(_modelHandle);
-            _modelHandle = IntPtr.Zero;
+            if (_modelHandle != IntPtr.Zero)
```

Comment:
Just out of curiosity, what problem was the existing code giving?

Reply:
ok, I see that it would hit if the _modelHandle was not created correctly. Thanks for the fix.

Reply:
Very rare case, as the initialization on the ctor would have to had failed, but I did hit this once and it crashed the app. Its a simple fix, so I rather not crash the whole app and fail gracefully here.